otx/cli/utils/hpo.py

```diff
@@ -70,766 +56,827 @@ def _check_hpo_enabled_task(task_type: TaskType):
     ]
 
 
-def _is_mpa_framework_task(task_type: TaskType):
-    """Check whether mpa framework task."""
-    return _is_cls_framework_task(task_type) or _is_det_framework_task(task_type) or _is_seg_framework_task(task_type)
+def check_hpopt_available() -> bool:
+    """Check whether hpopt is avaiable."""
+    return hpopt is not None
 
 
-def _is_cls_framework_task(task_type: TaskType):
-    """Check whether classification framework task."""
-    return task_type == TaskType.CLASSIFICATION
+class TaskManager:
+    """Task utility class to give common interface from different task.
 
+    Args:
+        task_type (TaskType): otx task type
+    """
 
-def _is_det_framework_task(task_type: TaskType):
-    """Check whether detection framework task."""
-    return task_type in [
-        TaskType.DETECTION,
-        TaskType.INSTANCE_SEGMENTATION,
-        TaskType.ROTATED_DETECTION,
-    ]
+    def __init__(self, task_type: TaskType):
+        self._task_type = task_type
 
+    @property
+    def task_type(self):
+        """Task_type property."""
+        return self._task_type
 
-def _is_seg_framework_task(task_type: TaskType):
-    """Check whether segmentation framework task."""
-    return task_type == TaskType.SEGMENTATION
+    def is_mmcv_framework_task(self) -> bool:
+        """Check task is run on mmcv.
 
+        Returns:
+            bool: whether task is run on mmcv
+        """
+        return self.is_cls_framework_task() or self.is_det_framework_task() or self.is_seg_framework_task()
 
-def _is_anomaly_framework_task(task_type: TaskType):
-    """Check whether anomaly framework task."""
-    return task_type in [
-        TaskType.ANOMALY_CLASSIFICATION,
-        TaskType.ANOMALY_DETECTION,
-        TaskType.ANOMALY_SEGMENTATION,
-    ]
+    def is_cls_framework_task(self) -> bool:
```

Comment:
This is the design choice, maybe we can gathering `is_cls_framework_task` `is_det_framework_task` `is_seg_framework_task`, ... to `get_task_type`?
``` python
get_task_type(task_type:TaskType):
    if task_type == TaskType.CLASSIFICATION:
         return 
```

Reply:
Actually, I think current implementation is easier to read. How about let it be?

Reply:
No problems :)